spec/Candid.md

```diff
@@ -1188,11 +1200,15 @@ M : <val> -> <reftype> -> i8*
 M(ref(r) : service <actortype>) = i8(0)
 M(id(v*) : service <actortype>) = i8(1) M(v* : vec nat8)
 
-M(ref(r)   : func <functype>) = i8(0)
-M(pub(s,n) : func <functype>) = i8(1) M(s : service {}) M(n : text)
+M(ref(r)         : func <functype>) = i8(0)
+M(pub(s,n)       : func <functype>) = i8(1) M(s : service {}) M(n : text)
+M(pub(s,n,v*:t*) : func <functype>) = i8(2) M(s : service {}) M(n : text) M*(v* : t*)
 
 M(ref(r) : principal) = i8(0)
 M(id(v*) : principal) = i8(1) M(v* : vec nat8)
+
+M* : <val>* -> <datatype>* -> i8*
+M*(v^N : <datatype>^N) = leb128(N) M(v : <datatype>)^N
```

Comment:
Where is the type of `v` encoded? I’d expect something like
```
 M*(v^N : <datatype>^N) = leb128(N) I(<datatype>^N)^N  M(v : <datatype>)^N
```
to match what we do in `B`